src/docs/stan-reference/appendices.tex

```diff
@@ -663,35 +664,55 @@ \subsection{Expressions}
 \small
 \begin{Verbatim}[fontsize=\small]
 expressions ::= expression % ','
-expression ::= real_literal
-             | variable
-             | '{' expressions '}'
-             | expression `?` expression `:` expression
+
+expression ::= expression `?` expression `:` expression
              | expression infixOp expression
              | prefixOp expression
              | expression postfixOp
-             | expression '[' indexes ']'
-             | function_literal '(' ?expressions ')'
-             | function_literal '(' expression ?('|' expression % ',') ')'
-             | integrate_ode '(' function_literal (',' expression){6} ')'
-             | integrate_ode_rk45
-               '(' function_literal (',' expression){6|9} ')'
-             | integrate_ode_bdf
-               '(' function_literal (',' expression){6|9} ')'
-             | algebra_solver
-             '('function_literal (',' expression){4|7} ')'
-             | '(' expression ')'
+             | common_expression
+
+range_expression ::= range_expression mathInfixOp range_expression
```

Comment:
There is no restriction on what can be a range expression other than that integers require integers and that all variables have to be have been introduced.  The only reason things like odes aren't allowed is that they don't return scalars.  So I think this is easier to do with a side condition than in the grammar.

One thing that's definitely missing here is functions, which are allowed in ranges.

Reply:
But what about [expression07_grammar_def.hpp](https://github.com/stan-dev/stan/blob/9d30c0be79f73c4d7a4d1542fce304ba1935f314/src/stan/lang/grammars/expression07_grammar_def.hpp).  Any operataor with lower precedence than `+`|`-` can't be inside a range expression in order to resolve the ambiguity of `<` and `>`.

Reply:
The inclusion of the `common_expression` rule in `range_expression` means that includes function expressions.